lib/logstash/inputs/file.rb

```diff
@@ -160,26 +164,71 @@ class LogStash::Inputs::File < LogStash::Inputs::Base
   # The default of 4095 is set in filewatch.
   config :max_open_files, :validate => :number
 
+  # What mode do you want the file input to operate in.
+  # Tail a few files or read many content-complete files
+  # The default is tail
+  # If "read" is specified then the following other settings are ignored
+  #   `start_position` (files are always read from the beginning)
+  #   `delimiter` (files are assumed to use \n or \r (or both) as line endings)
+  #   `close_older` (files are automatically 'closed' when EOF is reached)
+  # If "read" is specified then the following settings are heeded
+  #   `ignore_older` (older files are not processed)
+  # "read" mode now supports gzip file processing
+  config :mode, :validate => [ "tail", "read"], :default => "tail"
+
+  # When in 'read' mode, what action should be carried out when a file is done with.
+  # If 'delete' is specified then the file will be deleted.
+  # If 'log' is specified then the full path of the file is logged to the file specified
+  # in the `file_completed_log_path` setting.
+  config :file_completed_action, :validate => ["delete", "log"], :default => "delete"
+
+  # Which file should the completely read file paths be appended to.
+  # Only specify this path to a file when `file_completed_action` is 'log'.
+  config :file_completed_log_path, :validate => :string
+
+  # The sincedb entry now has a last active timestamp associated with it.
+  # If no changes are detected in tracked files in the last N days their sincedb
+  # tracking record will expire and not be persisted.
+  # This option protects against the well known inode recycling problem. (add reference)
+  config :sincedb_clean_after, :validate => :number,  :default => 14 # days
+
+  # This option sets up a form of striped read. Some lines from each active file will
+  # appear be processed in parallel.
+  # Given that the sort order of discovered files is OS dependent, it is not possible to control
+  # the file processing order. Each read iteration will process a 32KB chunk,
+  # e.g. 32 read iterations will process 1MB from each active file.
+  # See the option `max_open_files` for more info, but note in "read" mode, close_older is
+  # not required as files are closed at EOF
+  # The default set internally is very large (the maximum positive Ruby integer)
+  config :read_iterations, :validate => :number,  :default => (2**(0.size * 8 - 2) - 1)
```

Comment:
I dont understand this option - maybe we should rephrase it.

Reply:
for me both I guess, but if we have a good description then maybe the setting name will make more sense to me 😆